res/controllers/Denon-HC1000S-scripts.js

```diff
@@ -0,0 +1,869 @@
+                engine.setValue(group, "loop_scale", scaleFactor);
+            } else {
+                //move loop to right
+                DenonHC1000S.moveLoopRight(midino, control, value, status, group);
+                print("moving loop right");
+            }
+        } else { //make scale smaller (left turn)
+            if (!DenonHC1000S.resizeLoopButtonPressed) {
+                scaleFactor = (scaleFactor - scaleFactor * 1 / 2);
+                print("scale down by factor " + scaleFactor);
+                engine.setValue(group, "loop_scale", scaleFactor);
+            } else {
+                //move loop to left
+                DenonHC1000S.moveLoopLeft(midino, control, value, status, group);
+                print("moving loop left");
+            }
+        }
+    } else { //shift pressed -> change beatloop size with wheel
+        var beatloopSizeFactor = engine.getValue(group, "beatloop_size");
+        if (knobDelta > 0) { //make beatloop_size bigger (right turn)
+            if ((DenonHC1000S.resizeLoopButtonPressed) && (beatloopSizeFactor > 1)) {
+                scaleFactor = Math.round((beatloopSizeFactor + 1)); //single increment
+            } else {
+                scaleFactor = (beatloopSizeFactor * 2); //double
+                if (scaleFactor > 1) {
+                    scaleFactor = Math.round(scaleFactor);
+                }
+            }
+            print("change beatloop_size by factor " + scaleFactor);
+            engine.setValue(group, "beatloop_size", scaleFactor);
+        } else { //make beatloop_size  smaller (left turn)
+            if ((DenonHC1000S.resizeLoopButtonPressed) && (beatloopSizeFactor > 1)) {
+                scaleFactor = Math.round((beatloopSizeFactor - 1));
+            } else {
+                scaleFactor = (beatloopSizeFactor * 1 / 2);
+                if (scaleFactor > 1) {
+                    scaleFactor = Math.round(scaleFactor);
+                }
+            }
+            print("change beatloop_size by factor " + scaleFactor);
+            engine.setValue(group, "beatloop_size", scaleFactor);
+        }
+    }
+};
+
+DenonHC1000S.moveLoopLeft = function(midino, control, value, status, group) {
+    var samplesPerBeat = Math.floor(0.5 + DenonHC1000S.samplesPerBeat(group));
+    var loopInPosition = engine.getValue(group, "loop_start_position");
+    var loopOutPosition = engine.getValue(group, "loop_end_position");
+    if (loopInPosition !== -1 && loopOutPosition !== -1 && loopInPosition > samplesPerBeat) {
+        loopInPosition = loopInPosition - samplesPerBeat;
+        loopInPosition = Math.max(0, loopInPosition - loopInPosition % 2);
+        loopOutPosition = loopOutPosition - samplesPerBeat;
+        loopOutPosition = Math.max(0, loopOutPosition - loopOutPosition % 2);
+        engine.setValue(group, "loop_start_position", loopInPosition);
+        engine.setValue(group, "loop_end_position", loopOutPosition);
+    }
```

Comment:
You can probably replace this with some variant of [`loop_move_X_backward`](https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-loop_move_X_forward)